ethcore/src/state.rs

```diff
@@ -39,41 +39,97 @@ pub struct ApplyOutcome {
 pub type ApplyResult = Result<ApplyOutcome, Error>;
 
 #[derive(Debug)]
-enum AccountEntry {
+enum AccountData {
 	/// Contains account data.
 	Cached(Account),
 	/// Account has been deleted.
-	Killed,
 	/// Account does not exist.
 	Missing,
 }
 
-impl AccountEntry {
-	fn is_dirty(&self) -> bool {
+impl AccountData {
+	/// Clone dirty data into a new `AccountData`.
+	fn clone_dirty(&self) -> AccountData {
 		match *self {
-			AccountEntry::Cached(ref a) => a.is_dirty(),
-			AccountEntry::Killed => true,
-			AccountEntry::Missing => false,
+			AccountData::Cached(ref acc) => AccountData::Cached(acc.clone_dirty()),
+			AccountData::Missing => AccountData::Missing,
 		}
 	}
 
+	/// Clone account entry data that needs to be saved in the snapshot.
+	/// This includes basic account information and all locally modified storage keys
+	fn clone_for_snapshot(&self) -> AccountData {
```

Comment:
How is that different from `clone_dirty`?

Perhaps should use `clone_for_snapshot`?
Kind of indicates that it might be untested too.